### PR TITLE
Photo Capture Implementations

### PR DESCRIPTION
**Pull request by KalaiselvanK-018**

### CODE REVIEW

File: survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java [hunk 1]
```diff
@@ -158,20 +203,52 @@ public void onResponse(String response) {
...
         closeButton.setBackgroundResource(R.drawable.ic_close_rounded);
         closeButton.setClickable(true);
         closeButton.setPadding(10, 10, 10, 10);
         closeButton.setVisibility(View.GONE);
 
-        FrameLayout.LayoutParams closeButtonParams = new FrameLayout.LayoutParams(
-            FrameLayout.LayoutParams.WRAP_CONTENT,
-            FrameLayout.LayoutParams.WRAP_CONTENT
+        ConstraintLayout.LayoutParams closeButtonParams = new ConstraintLayout.LayoutParams(
+                ConstraintLayout.LayoutParams.WRAP_CONTENT,
```
Comment (Gokulkrishna-Raju) [inline survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java:220]: @KalaiselvanK-018 ,  Any specific reason to change it from FrameLayout to ConstraintLayout ?

File: survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java [hunk 2]
```diff
@@ -158,20 +203,52 @@ public void onResponse(String response) {
...
+        closeButtonParams.topToTop = ConstraintLayout.LayoutParams.PARENT_ID;
+        closeButtonParams.setMargins(0, marginInPx, marginInPx, 0);
+
         closeButton.setLayoutParams(closeButtonParams);
+        constraintLayout.addView(closeButton);
+
+
+        HashMap properties = survey.getProperties();
+
+
```
Comment (Gokulkrishna-Raju) [inline survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java:239]: Remove the unnecessary Spaces

File: survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java [hunk 3]
```diff
@@ -147,7 +192,7 @@ public void onResponse(String response) {
             }
         }
 
-        final FrameLayout ssLayout = new FrameLayout(Objects.requireNonNull(getActivity()));
```
Comment (Gokulkrishna-Raju) [inline survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java:150]: Does requireActivity always gives a non null activity ?

File: survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java [hunk 4]
```diff
@@ -158,20 +203,52 @@ public void onResponse(String response) {
...
+
+
+        HashMap properties = survey.getProperties();
+
+
+
+
+
+
+        ssWebView.setWebViewClient(new WebViewClient() {
```
Comment (Gokulkrishna-Raju) [inline survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java:244]: @KalaiselvanK-018 

ssWebView.setWebViewClient(new WebViewClient() {
            @Override
            public void onPageFinished(WebView view, String url) {
                super.onPageFinished(view, url);
                String jsCode = "const styleTag = document.createElement('style'); styleTag.innerHTML = `.ss-language-selector--wrapper { margin-right: 45px; }`; document.body.appendChild(styleTag);";
                view.evaluateJavascript(jsCode, null);
            }
        });


this code seems to be duplicate, can we check it ?

File: survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java [hunk 5]
```diff
@@ -237,17 +316,30 @@ public void onPageFinished(WebView view, String url) {
 
             @Override
             public boolean onShowFileChooser(WebView webView, ValueCallback<Uri[]> filePathCallback, FileChooserParams fileChooserParams) {
+
+                Log.d("msg","isCaptureImageActive="+isCaptureImageActive);
```
Comment (Gokulkrishna-Raju) [inline survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java:320]: Remove debug logs

File: survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java [hunk 6]
```diff
@@ -237,17 +316,30 @@ public void onPageFinished(WebView view, String url) {
...
-                Intent intent = fileChooserParams.createIntent();
-                try {
-                    startActivityForResult(intent, REQUEST_SELECT_FILE);
-                } catch (ActivityNotFoundException e) {
-                    mUploadMessageArray = null;
-                    Toast.makeText(Objects.requireNonNull(getActivity()).getApplicationContext(), "Cannot open file chooser", Toast.LENGTH_LONG).show();
-                    return false;
+
+                if(isCaptureImageActive){
+
```
Comment (Gokulkrishna-Raju) [inline survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java:329]: remove unnecessary spaces here too

File: survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java [hunk 7]
```diff
@@ -268,31 +360,96 @@ public void onProgressChanged(WebView view, int newProgress) {
...
         public void shareData(String data) {
             surveyCompleted = true;
             onSsResponseEventListener.onSsResponseEvent(SurveySparrow.toJSON(data));
         }
+
+        @JavascriptInterface
+        public void captureImage() {
+
+            if (!isCaptureImageActive) {
+                isCaptureImageActive = true;
```
Comment (Gokulkrishna-Raju) [inline survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java:430]: format the function (spaces and intent)

File: survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java [hunk 8]
```diff
@@ -268,31 +360,96 @@ public void onProgressChanged(WebView view, int newProgress) {
...
         return ssLayout;
     }
 
     @Override
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         if (requestCode == REQUEST_SELECT_FILE) {
             if (mUploadMessageArray == null) return;
-            mUploadMessageArray.onReceiveValue(WebChromeClient.FileChooserParams.parseResult(resultCode, data));
+            Uri[] results = WebChromeClient.FileChooserParams.parseResult(resultCode, data);
+            Log.d("REQUEST_SELECT_FILE", "results: " + (results != null ? Arrays.toString(results) : "null"));
```
Comment (Gokulkrishna-Raju) [inline survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java:374]: Remove debug logs

File: survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java (hunk 2 above)
Comment (KalaiselvanK-018) [reply survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java:239]: ok

File: survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java (hunk 3 above)
Comment (KalaiselvanK-018) [reply survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java:150]: i had already verified.

File: survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java [hunk 9]
```diff
@@ -268,31 +360,96 @@ public void onProgressChanged(WebView view, int newProgress) {
...
+            mUploadMessageArray.onReceiveValue(results);
             mUploadMessageArray = null;
         } else if (requestCode == FILE_CHOOSER_RESULT_CODE) {
-            if (null == mUploadMessage) return;
+            if (mUploadMessage == null) return;
             Uri result = data == null || resultCode != Activity.RESULT_OK ? null : data.getData();
+            Log.d("FILE_CHOOSER_RESULT_CODE", "result: " + (result != null ? result.toString() : "null"));
```
[the remark this answers is not in the document]
Comment (KalaiselvanK-018) [reply survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java:380]: it is already there bro..

File: survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java (hunk 8 above)
Comment (KalaiselvanK-018) [reply survey-sparrow-android-sdk/src/main/java/com/surveysparrow/ss_android_sdk/SsSurveyFragment.java:374]: it is already there bro..

**Pull request merged**